### PR TITLE
[CN-373] Separate E2E test suite into multiple suites

### PR DESCRIPTION
**Pull request by mtyazici**
Changes:
- Separated the test suite into multiple files. Every file is now a suite. There are following suites: `hz`, `mc`, `hz_persistence`, `hz_expose_externally`, `map`, `map_persistence`.
- Suites are defined with labels. You can run single or multiple suites with following commands. `make test-e2e E2E_TEST_SUITE=hz || mc`. To run all test suites just run `make test-e2e`.
- I added `-p` option to Ginkgo CLI. This will find the core count of the host and create as many processes to parallelize the tests. This somehow limits the parallelization if the core count is limited. This is the case for our GitHub actions hosted runners, which have 2 cores. 

I tried to run all test suites in matrices in the workflow. However, I got unexpected errors. If we want to improve the parallelization, we should look more into using matrices in the workflows.

### CODE REVIEW

File: test/e2e/hazelcast_exposeExternally_test.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
+package e2e
```
Comment (SeriyBg) [inline test/e2e/hazelcast_exposeExternally_test.go:1]: Is there a reason why the file name is a mix of snake-case and camel-case?

File: test/e2e/hazelcast_test.go [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+	hazelcastcomv1alpha1 "github.com/hazelcast/hazelcast-platform-operator/api/v1alpha1"
+	hazelcastconfig "github.com/hazelcast/hazelcast-platform-operator/test/e2e/config/hazelcast"
+)
+
+const (
+	logInterval = 10 * time.Millisecond
+)
+
+var _ = FDescribe("Hazelcast", Label("hz"), func() {
```
Comment (SeriyBg) [inline test/e2e/hazelcast_test.go:23]: Here as well should be `Describe`?

File: test/e2e/hazelcast_exposeExternally_test.go (hunk 1 above)
Comment (mtyazici) [reply test/e2e/hazelcast_exposeExternally_test.go:1]: Not particularly, I will update it. Thanks!

**Review by cagric0 — APPROVED** [review]
Thanks for the good job. LGTM!

**Review by SeriyBg — APPROVED** [review]
Other than a small nit, LGTM

**Pull request merged**